custom-components/react-carousel.js

```diff
@@ -0,0 +1,114 @@
+let UID = 0;
+function newId() {
+    return UID++;
+}
+
+const e = React.createElement;
+
+function Product(props) {
+    const record = props.record;
+    return e('a', {href: record.Deeplink, className: 'product'},
+        e('img', {src: record.ImageURL, alt: record.Title}),
+        e('div', {className: 'title'}, record.Title),
+        e('div', {className: 'price'}, record.Price + ' €')
+    );
+}
+
+function Carousel(props) {
+    const ref = React.useRef(null);
+
+    const offsetState = React.useState(0);
+    const offset = offsetState[0], setOffset = offsetState[1];
```

Comment:
Optional:
Convention I've been learning so far is to use `setState` in following manner
`import { useState} from 'react';`
`const [offset, setOffset] = setState(0) `

Reply:
Ah ok, I didn't notice that you are not using webpack in this example.

Its hard for me to tell if we should keep both versions. Converting to classes shouldn't be too difficult though but there's always someone who do copy&paste without trying to understand what's inside. I'll leave it up to you. In my opinion it could stay with hooks which should be used more often in the future.